src/snapshot.rs

```diff
@@ -89,26 +89,37 @@ pub enum FunctionCodeHandling {
 /// Helper class to create a snapshot data blob.
 #[repr(C)]
 #[derive(Debug)]
-pub struct SnapshotCreator([usize; 1]);
+pub(crate) struct SnapshotCreator([usize; 1]);
 
 impl SnapshotCreator {
   /// Create and enter an isolate, and set it up for serialization.
   /// The isolate is created from scratch.
   #[inline(always)]
-  pub fn new(external_references: Option<&'static ExternalReferences>) -> Self {
+  #[allow(clippy::new_ret_no_self)]
+  pub(crate) fn new(
+    external_references: Option<&'static ExternalReferences>,
+  ) -> OwnedIsolate {
     let mut snapshot_creator: MaybeUninit<Self> = MaybeUninit::uninit();
     let external_references_ptr = if let Some(er) = external_references {
       er.as_ptr()
     } else {
       std::ptr::null()
     };
-    unsafe {
+    let snapshot_creator = unsafe {
       v8__SnapshotCreator__CONSTRUCT(
         &mut snapshot_creator,
         external_references_ptr,
       );
       snapshot_creator.assume_init()
-    }
+    };
+
+    let isolate_ptr =
+      unsafe { v8__SnapshotCreator__GetIsolate(&snapshot_creator) };
+    let mut owned_isolate = OwnedIsolate::new(isolate_ptr);
+    ScopeData::new_root(&mut owned_isolate);
+    owned_isolate.create_annex(Box::new(()));
+    owned_isolate.set_slot(snapshot_creator);
+    owned_isolate
```

Comment:
This is needed for https://github.com/denoland/rusty_v8/pull/973 where we will store allocations of existing blob